src/_gettsim/transfers/rente.py

```diff
@@ -19,88 +19,224 @@ def sum_ges_rente_priv_rente_m(priv_rente_m: float, ges_rente_m: float) -> float
-    following equation:
+    """Calculate total individual public pension including Grundrentenzuschlag.
 
-    .. math::
+    Parameters
+    ----------
+    bruttorente_m
+        See :func:`bruttorente_m`.
+    grundr_zuschlag_m
+        See :func:`grundr_zuschlag_m`.
+    rentner
+        See basic input variable :ref:`rentner <rentner>`.
 
-        R = EP * ZF * Rw
+    Returns
+    -------
 
-    models 'Rentenformel':
-    https://de.wikipedia.org/wiki/Rentenformel
-    https://de.wikipedia.org/wiki/Rentenanpassungsformel
+    """
+    out = bruttorente_m + grundr_zuschlag_m if rentner else 0.0
+    return out
+
+
+@policy_info(
+    end_date="2016-12-31",
+    name_in_dag="bruttorente_m",
+    params_key_for_rounding="ges_rente",
+)
+def bruttorente_mit_harter_hinzuverdienstgrenze_m(
+    alter: int,
+    ges_rente_regelaltersgrenze: float,
+    bruttolohn_m: float,
+    bruttorente_basisbetrag_m: float,
+    ges_rente_params: dict,
+) -> float:
+    """Pension benefits after earnings test for early retirees.
 
+    If earnings are above an earnings limit, the pension is fully deducted.
 
     Parameters
     ----------
-    ges_rente_zugangsfaktor
-        See :func:`ges_rente_zugangsfaktor`.
-    entgeltp_ost_update
-        See :func:`entgeltp_ost_update`.
-    entgeltp_west_update
-        See :func:`entgeltp_west_update`.
-    rentner
-        See basic input variable :ref:`rentner <rentner>`.
+    alter
+        See basic input variable :ref:`alter <alter>`.
+    ges_rente_regelaltersgrenze
+        See :func:`ges_rente_regelaltersgrenze`.
+    bruttolohn_m
+        See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    bruttorente_basisbetrag_m
+        See :func:`bruttorente_basisbetrag_m`.
     ges_rente_params
         See params documentation :ref:`ges_rente_params <ges_rente_params>`.
 
     Returns
     -------
 
     """
-
-    if rentner:
-        out = (
-            entgeltp_west_update * ges_rente_params["rentenwert"]["west"]
-            + entgeltp_ost_update * ges_rente_params["rentenwert"]["ost"]
-        ) * ges_rente_zugangsfaktor
+    # TODO (@MImmesberger): Use age with monthly precision.
+    # https://github.com/iza-institute-of-labor-economics/gettsim/issues/781
+    if (alter >= ges_rente_regelaltersgrenze) or (
+        bruttolohn_m <= ges_rente_params["hinzuverdienstgrenze"] / 12
+    ):
+        out = bruttorente_basisbetrag_m
     else:
         out = 0.0
 
     return out
 
 
-@policy_info(end_date="2020-12-31")
-def ges_rente_m(ges_rente_vor_grundr_m: float) -> float:
-    return ges_rente_vor_grundr_m
+@policy_info(
+    start_date="2017-01-01",
+    end_date="2022-12-31",
+    name_in_dag="bruttorente_m",
+    params_key_for_rounding="ges_rente",
+)
+def bruttorente_mit_hinzuverdienstdeckel_m(  # noqa: PLR0913
+    alter: int,
+    ges_rente_regelaltersgrenze: float,
+    bruttorente_basisbetrag_m: float,
+    bruttolohn_m: float,
```

Comment:
```suggestion
    bruttolohn_y: float,
```

Should be the same in the function, maybe a bit more difficult here, maybe even postpone via a to-do (we will be able to do that for params eventually, too). But we should not have any time unit conversions within functions.

Reply:
I think we can go with `bruttolohn_y` here and avoid parameter conversions. This just requires to split the function up into several functions (because we need automatic time conversions across them); will do this soon.